### PR TITLE
Add 403 error response to federation/v1/state_ids

### LINKED ISSUE #1912: Add 403 response error to `/_matrix/federation/v1/state_ids/{roomId}`
**Suggestion**
[`/_matrix/federation/v1/state_ids/{roomId}`](https://spec.matrix.org/v1.11/server-server-api/#get_matrixfederationv1state_idsroomid) currently specifies `200` and `404` status code responses. However, homeserver implementations ([Synapse](https://github.com/element-hq/synapse/blob/7d8f0ef351e99adf602b3acb67b2516a02ff6918/synapse/federation/federation_server.py#L605-L607), [Conduit](https://gitlab.com/famedly/conduit/-/blob/next/src/api/server_server.rs#L1447-1456)) have been returning `403`s in the following cases:

1. The host is not in the room.
2. The host is banned from the room via its ACL.

403 should be specified as a possible response code in these cases.

### PR DESCRIPTION
**Pull request by AllMightLegend**
fixes #1912 by adding a 403 error response to /_matrix/federation/v1/state_ids/{roomId}

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Fixes issue #1912 by adding error response to /_matrix/federation/v1/state_ids/{roomId}

Signed-off-by: `Srinjoy Sen Chowdhury  srinjoysen123@gmail.com`

Preview: https://pr1926--matrix-spec-previews.netlify.app

### CODE REVIEW

**Review by AllMightLegend — COMMENTED** [review]
Changed directory

File: changelogs/server_server/newsfragments/1926.new [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: ../client-server/definitions/errors/error.yaml
+              examples:
+                response:
+                  value: {
+                    "errcode": "M_FORBIDDEN",
+                    "error": "You do not have permission to access the state at the requested event."
+                  }
```
[the remark this answers is not in the document]
Comment (richvdh) [reply changelogs/server_server/newsfragments/1926.new:18]: Have you looked at other examples? Start there. If you still have questions, come back.

[the remark this answers is not in the document]
Comment (AllMightLegend) [reply changelogs/server_server/newsfragments/1926.new:18]: okay thank you sir

File: changelogs/server_server/newsfragments/1926.new [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: ../client-server/definitions/errors/error.yaml
+              examples:
+                response:
+                  value: {
+                    "errcode": "M_FORBIDDEN",
+                    "error": "You do not have permission to access the state at the requested event or you have been banned."
+                  }
```
Comment (richvdh) [inline changelogs/server_server/newsfragments/1926.new:18]:
```suggestion
Add 403 error response to `/_matrix/federation/v1/state_ids/{roomId}`
```

**Review by richvdh — APPROVED** [review]

**Comment by AllMightLegend** [comment]
Hello Sir,I would like to contribute more and aim for GSOC.Would you be able to help me ?

**Comment by AllMightLegend** [comment]
@richvdh Sir I would like to participate in GSOC. Would you be able to help me ??

**Comment by richvdh** [comment]
I don't think the Matrix project has any plans to participate in GSOC, but in any case you'll need to complete the application process. Try https://summerofcode.withgoogle.com/.

**Pull request merged**